src/index.js

```diff
@@ -0,0 +1,284 @@
+    this.pluginConfig = this.embark.pluginConfig;
+    this.deviceIp = this.pluginConfig.deviceIp;
+    this.logger = this.embark.logger;
+    // we want a timeout to occur before issuing the next request, so reduce the timeout specified
+    this.statusApi = new StatusApi({ip: this.deviceIp, logger: this.logger, timeout: CONNECT_INTERVAL - 1000});
+    this.webServerConfig = {};
+    this.blockchainConfig = {};
+    this.networkIdDirectory = path.join(process.env.DAPP_PATH, '.embark', 'embark-status');
+    this.networkIdPath = path.join(this.networkIdDirectory, 'networkId');
+
+    // gets hydrated webserver config from embark
+    this.events.on('config:load:webserver', webServerConfig => {
+      this.webServerConfig = webServerConfig;
+    });
+
+    // gets hydrated blockchain config from embark
+    this.events.on('config:load:blockchain', blockchainConfig => {
+      this.blockchainConfig = blockchainConfig;
+    });
+
+    // adds cors to blockchain and storage clients
+    this.events.request('config:cors:add', `http://${this.deviceIp}`);
+
+    // register service check
+    this[_registerEvents]();
+
+    // kick off network connection, and open dapp after connection
+    this.connectStatusToEmbark();
+  }
+
+  /**
+   * Continuously attempts to connect to the Status app by instructing
+   * the Status app to connect to this dApp's network (node) until a 
+   * successful connection is made. If the network doesn't exist, it is
+   * added. Once a connection is made, the Status app is instructed to
+   * open this dApp in it's dApp browser.
+   * 
+   * @returns {void}
+   */
+  connectStatusToEmbark() {
+    // keep trying to connect to the status app if previous connections were unsuccessful
+    this.events.once('web3Ready', (_isReady) => {
+      this.events.request('blockchain:networkId', networkId => {
+        this.networkId = networkId;
+        const connectIntervalId = setInterval(() => {
+          this[_connectStatus](err => {
+            if (err) return this.logger.error(err);
+            clearInterval(connectIntervalId);
+            this.openDapp(err => {
+              if (err) this.logger.error(err);
+            });
+          });
+        }, CONNECT_INTERVAL);
+      });
+    });
+  }
+
+  /**
+   * Opens the Embark dApp in the Status dApp browser.
+   * 
+   * @param {Function} cb Callback called after the open dApp command is sent 
+   * to the Status app, called with parameters:
+   *  - {String} err Error that occurred opening the dApp in the Status
+   *                 browser.
+   *  - {Object} result Result returned from the Status app on successful 
+   *             opening of the dApp.
+   * 
+   * @returns {void}
+   */
+  openDapp(cb) {
+    let dappHost = this.webServerConfig.host;
+    if (dappHost === '0.0.0.0' || dappHost === 'localhost') {
+      dappHost = Ip.address();
+    }
+
+    const dappUrl = this.pluginConfig.dappUrl || buildUrl('http', dappHost, this.webServerConfig.port) + '/';
```

Comment:
Similar as in the other place, can we use http here all the time without this going bananas with https?

Reply:
I think this is dependent upon the webserver connection on the app, and as far as I know this is `http` only for now. I guess man-in-the-middle attacks would be difficult given that the comms are localised to a local network (the connection is from a host machine running embark to a phone on the same LAN, and the HTTP comms go from machine to phone). I would say this is OK for now.

Reply:
Thanks for clarifying!